src/RapiSender.cpp

```diff
@@ -0,0 +1,227 @@
+#include <Arduino.h>
+#include "RapiSender.h"
+
+//#define DBG
+
+#ifdef DBG
+#include <SoftwareSerial.h>
+extern SoftwareSerial alt;
+#define dbgprint(s) alt.print(s)
+#define dbgprintln(s) alt.println(s)
+#else
+#define dbgprint(s)
+#define dbgprintln(s)
+#endif // DBG
+
+// convert 2-digit hex string to uint8_t
+uint8_t
+htou8(const char *s) {
+  uint8_t u = 0;
+  for (int i = 0; i < 2; i++) {
+    char c = s[i];
+    if (c != '\0') {
+      if (i == 1)
+        u <<= 4;
+      if ((c >= '0') && (c <= '9')) {
+        u += c - '0';
+      } else if ((c >= 'A') && (c <= 'F')) {
+        u += c - 'A' + 10;
+      }
+      //      else if ((c >= 'a') && (c <= 'f')) {
+      //        u += c - 'a' + 10;
+      //      }
+      else {
+        // invalid character received
+        return 0;
+      }
+    }
+  }
+  return u;
+}
+
+
+RapiSender::RapiSender(Stream * stream) {
+  _stream = stream;
+  *_respBuf = 0;
+  _flags = 0;
+
+  _sequenceId = RAPI_INVALID_SEQUENCE_ID;
+}
+
+// return = 0 = OK
+//        = 1 = command will cause buffer overflow
+void
+RapiSender::_sendCmd(const char *cmdstr) {
+  _stream->print(cmdstr);
+  dbgprint(cmdstr);
+
+  const char *s = cmdstr;
+  uint8_t chk = 0;
```

Comment:
Consider putting this in a (inline) method as it is duplicated a few times and a well named method will also help readability